client/src/pages/SearchMovie/index.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useCallback } from 'react'
+
+import { SearchMovieContainer } from './styled'
+import { BebasText } from '../../styles/Text'
+import { SearchBar, SearchInput, SearchImg } from '../../styles/SearchBar'
+import { useConfiguration } from '../../providers/Configuration'
+import { authFetch } from '../../auth'
+import debounce from 'lodash.debounce'
+import Result from './Results'
+
+const SearchMovie = () => {
+  const { apiUrl } = useConfiguration()
+  const [movieResults, setMovieResult] = useState([])
+
+  const debouncedSave = debounce(
+    (currentValue) => searchMovies(currentValue),
+    1000
+  )
+
+  async function searchMovies(currentValue) {
+    if (!currentValue) {
+      setMovieResult([])
+      return
+    }
+    try {
+      const response = await authFetch(
+        `${apiUrl}/movies?search=${currentValue}`
+      )
+      const data = await response.json()
+      if (data.Search) {
+        setMovieResult(data.Search)
+      }
+    } catch (error) {
+      console.log('error', error)
+    }
+  }
+
+  const handleChange = (e) => {
+    const { value: currentValue } = e.target
```

Comment:
Probably not necessary to destructure and alias value, since you have no other variables in this function.
This could simply be 
```
const {value} = e.target
```

Or even
```
debouncedSave(e.target.value)
```